### PR TITLE
Call mappings with event fixtures

### PR DESCRIPTION
**Pull request by cargopete**
The "call mappings with event fixtures" functionality is already technically implemented in https://github.com/LimeChain/subgraph-tester/pull/4 , using the logic there and looping through an array of Event objects, passing them through the event handler function, results in properly hydrating the state. Tests were update to more clearly show this behaviour.

### CODE REVIEW

File: src/classes/Entity.ts [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
+  }
 
-    constructor(id: string, type: string, data: string) {
-        this.id = id;
-        this.type = type;
-        this.data = data;
-    }
+  public equals = (e: Entity) => {
+    return (
+      sha256(JSON.stringify(e)).toString() ===
```
Comment (vminkov) [inline src/classes/Entity.ts]: are we certain about JSON.stringify() doing deterministic ordered serialization?

**Review by vminkov — CHANGES_REQUESTED** [review]
to mock a contract call one should be able to set an event fixture as part of the mock call execution (and these fixtures are not necessarily the same for the same arguments list, each mock invocation should define it's own fixture of emitted events)

File: src/classes/Entity.ts (hunk 1 above)
Comment (cargopete) [reply src/classes/Entity.ts]: will have to read/test more on it

Comment (cargopete) [reply src/classes/Entity.ts]: I think we should go with https://www.npmjs.com/package/fast-json-stable-stringify

File: src/classes/Entity.ts [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
+  }
 
-    constructor(id: string, type: string, data: string) {
-        this.id = id;
-        this.type = type;
-        this.data = data;
-    }
+  public equals = (e: Entity) => {
+    return (
+      sha256(stringify(e)).toString() === sha256(stringify(this)).toString()
```
Comment (vminkov) [inline src/classes/Entity.ts:16]: https://www.mattzeunert.com/2016/01/28/javascript-deep-equal.html#:~:text=Deep%20equal%20is%20much%20more,stringify%20does%20not%20serialize%20functions.&text=deepEqual%20will%20instead%20check%20for,counterpart%20in%20the%20other%20object.

**Review by vminkov — APPROVED** [review]
stringify vs deep copy will be fixed later

**Pull request merged**